torch/fx/graph.py

```diff
@@ -101,10 +113,78 @@ def create_node(self, op: str, target: Target,
         self._mark_uses(args)
         self._mark_uses(kwargs)
         n = Node(self, name if name is not None else self._name(target), op, target, args, kwargs)
-        assert self._nodes[-1].op == 'return'
-        self._nodes.insert(-1, n)
+        self.insert_node_before(n=n, before=self._insert_point)
         return n
 
+    def insert_node_before(self, n : Node, before : Node) -> Node:
+        """
+        Insert node `n` directly before `before` in the topological ordering of nodes
+        """
+        # TODO: computationally inefficient
+        before_idx = self._nodes.index(before)
+        self._nodes.insert(before_idx, n)
+        return n
+
+    def find_all_uses_of(self, to_find : Node) -> List[Node]:
+        """
+        Find all nodes that use `to_find` in their args or kwargs
+        """
+        # TODO: computationally inefficient
+        def find_use(arg : Any) -> bool:
```

Comment:
The list of acceptable arguments should _only_ appear in a single place. Currently that is `map_arg`. Reproducing it here makes it really difficult to add other aggregates.

Reply:
We'll probably need something like `reduce_arg` then

Reply:
nvm, can hack this up with captures